lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -43,6 +43,32 @@ void SubstraitDialect::initialize() {
 #include "substrait-mlir/Dialect/Substrait/IR/SubstraitOpInterfaces.cpp.inc"
 #include "substrait-mlir/Dialect/Substrait/IR/SubstraitTypeInterfaces.cpp.inc"
 
+//===----------------------------------------------------------------------===//
+// Custom Parser and Printer for Substrait
+//===----------------------------------------------------------------------===//
+
+bool parseCountAsAll(OpAsmParser &parser, IntegerAttr &count) {
+  if (!parser.parseOptionalKeyword("all")) {
+    count = parser.getBuilder().getI64IntegerAttr(-1);
+  } else {
+    int64_t result;
+    if (!parser.parseInteger(result)) {
+      count = parser.getBuilder().getI64IntegerAttr(result);
+    } else {
+      return true;
+    }
+  }
+  return false;
```

Comment:
The control flow is a bit hard to understand here. It is often better to handle cases that return and continue without indentention/`else` branch:

```suggestion
  // `all` keyword (corresponds to `-1`).
  if (!parser.parseOptionalKeyword("all")) {
    count = parser.getBuilder().getI64IntegerAttr(-1);
    return false
  }
  
  // Normal integer.
  int64_t result;
  if (!parser.parseInteger(result)) {
    count = parser.getBuilder().getI64IntegerAttr(result);
    return false;
  }
  
  // Error.
  return true;
```